lib/large_object_store.rb

```diff
@@ -67,9 +67,9 @@ def read(key)
         # use values_at to enforce key order because read_multi doesn't guarantee a return order
         slices = @store.read_multi(*keys).values_at(*keys)
         return nil if slices.compact.size != pages
-        slices.map! { |s| [s.slice!(0, UUID_SIZE), s] }
+        slices = slices.map { |s| [s[0...UUID_SIZE], s[UUID_SIZE..-1]] }
         return nil unless slices.map(&:first).uniq == [uuid]
-        slices.map!(&:last).join("")
+        slices.map(&:last).join("")
```

Comment:
do we need to update them in-place if we are just returning and never reusing though? Just being safe when I think it makes sense.